[Ng Shi Jun] iP

ngshijun opened the pull request:
# LaMDA
```
  ██▓    ▄▄▄       ███▄ ▄███▓▓█████▄  ▄▄▄
  ▓██▒   ▒████▄    ▓██▒▀█▀ ██▒▒██▀ ██▌▒████▄
  ▒██░   ▒██  ▀█▄  ▓██    ▓██░░██   █▌▒██  ▀█▄
  ▒██░   ░██▄▄▄▄██ ▒██    ▒██ ░▓█▄   ▌░██▄▄▄▄██
  ░██████▒▓█   ▓██▒▒██▒   ░██▒░▒████▓  ▓█   ▓██▒
  ░ ▒░▓  ░▒▒   ▓▒█░░ ▒░   ░  ░ ▒▒▓  ▒  ▒▒   ▓▒█░
  ░ ░ ▒  ░ ▒   ▒▒ ░░  ░      ░ ░ ▒  ▒   ▒   ▒▒ ░
    ░ ░    ░   ▒   ░      ░    ░ ░  ░   ░   ▒
      ░  ░     ░  ░       ░      ░          ░  ░
                               ░
```
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes)

LaMDA frees your mind of having to remember things you need to do. It's,

- text-based
- eady to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/ngshijun/ip/releases/download/v0.1/LaMDA.jar).
2. double-click it.
3. add your tasks.
4. let it manage you tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

ngshijun committed:
Level-1

ngshijun committed:
remove unwanted texts

ngshijun committed:
Level-2, unable to detect duplicate input

ngshijun committed:
Level-3, no input check for index out of bound error for mark and unmark commands

ngshijun committed:
Level-4, no input check yet

ngshijun committed:
Exceptions handling

ngshijun committed:
Level-5

ngshijun committed:
Level-6

ngshijun committed:
Level-6

ngshijun committed:
Implement save and load script

ngshijun committed:
Implement date detection script for deadline task

ngshijun committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Implement date detection script for deadline task

# Conflicts:
#	src/main/java/Duke.java

ngshijun committed:
Resolve conflict on merge

ngshijun committed:
Merge pull request #1 from nus-cs2103-AY2223S1/add-gradle-support

Add Gradle support

ngshijun committed:
Resolve conflict

ngshijun committed:
Revert "Resolve conflict"

This reverts commit cea95be5710d437a55f1f1608d988b50eede32ec.

ngshijun committed:
Revert "Revert "Resolve conflict""

This reverts commit f6acbd1325300d2707ee5d38b9b6694baa3a794d.

ngshijun committed:
Implement A-MoreOOP

ngshijun committed:
Implement A-MoreOOP

ngshijun committed:
Implement packages

ngshijun committed:
Implement packages

ngshijun committed:
Add Gradle

ngshijun committed:
Implement JUnit test cases

ngshijun committed:
Refine input checker

ngshijun committed:
Adjust JUnit test cases

ngshijun committed:
Add find function

ngshijun committed:
remove unused variable

ngshijun committed:
Add JavaDoc to the code

ngshijun committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc to the code

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/task/Task.java

ngshijun committed:
Add JavaDoc for new methods

ngshijun committed:
define coding standard

ngshijun committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  define coding standard

# Conflicts:
#	src/main/java/duke/Ui.java

ngshijun committed:
Add JavaDocs for new methods

brendantwh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,157 @@
+
+public class Ui {
+
+    Scanner scn;
```
Should this be private?

ngshijun replied:
Yea, thanks for spotting it.

brendantwh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,111 @@
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DukeException("Argument of mark cannot be empty.");
+                } catch (NumberFormatException e) {
+                    throw new DukeException("The argument you've input is not an integer");
+                }
+            case "unmark":
+                try {
+                    int idx = Integer.parseInt(str[1]);
+                    try {
+                        return new UnmarkCommand(idx - 1);
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new DukeException("Invalid index.");
+                    }
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DukeException("Argument of unmark cannot be empty.");
+                } catch (NumberFormatException e) {
+                    throw new DukeException("The argument you've input is not an integer");
+                }
+            case "delete":
+                try {
+                    int idx = Integer.parseInt(str[1]);
+                    try {
+                        return new DeleteCommand(idx - 1);
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new DukeException("Invalid index.");
+                    }
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DukeException("Argument of delete cannot be empty.");
+                } catch (NumberFormatException e) {
+                    throw new DukeException("The argument you've input is not an integer");
+                }
+            case "find":
+                try {
+                    if (str[1].length() == 0) {
+                        throw new DukeException("The description of find cannot be empty.");
+                    }
+                    return new FindCommand(str[1]);
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DukeException("The description of find cannot be empty.");
+                }
```
Quite a bit of nesting here, maybe you could consider abstracting the code in each case into their relevant methods?

brendantwh reviewed (commented):
Mostly clean and readable code apart from the nesting issue highlighted. Good job!

ngshijun committed:
Uses GFMD in the PR description

ngshijun committed:
Reverts to a previous state

ngshijun committed:
Add CheckStyle support and modify the code according to the CheckStyle standard

ngshijun committed:
Add GUI to Duke

ngshijun committed:
Update GUI asthetic

ngshijun committed:
Update GUI visuals

tin-jy commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ * {@code Deadline} is a type of {@code Task} where user must specify the description and the "/by" remark
+ */
+public class Deadline extends Task {
+
```
Perhaps you want to rename Deadline to DeadlineTask to make it obvious that it is a Task?

tin-jy commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+     */
+    public Deadline(String description, LocalDate date) {
+        super(description);
+        this.date = date;
```
Perhaps this can be expanded to accept more variations of date inputs as well as to include time

tin-jy commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,14 @@
+mark 1
+list
+bye
+
```
You could consider adding some incorrect commands and see how Duke responds to them

tin-jy commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,41 @@
+/**
+ * Command to add an {@code Event} to the {@code TaskList}
+ */
+public class EventCommand extends Command {
```
Possibly the three Task commands can be further grouped

tin-jy reviewed (commented):
Looks nice!

ngshijun committed:
Add method to allow the program to exit automatically

ngshijun committed:
Update javaDocs

ngshijun committed:
Remove non-relevent test cases

ngshijun committed:
Merge branch 'branch-A-JavaDoc' of https://github.com/ngshijun/ip into branch-A-CodingStandard

* 'branch-A-JavaDoc' of https://github.com/ngshijun/ip:
  Add JavaDoc to the code

# Conflicts:
#	src/main/java/duke/Ui.java

ngshijun committed:
Merge branch 'branch-A-CodingStandard' of https://github.com/ngshijun/ip into branch-A-JavaDoc

* 'branch-A-CodingStandard' of https://github.com/ngshijun/ip:
  define coding standard

ngshijun committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:

# Conflicts:
#	src/main/java/duke/Ui.java

ngshijun committed:
Adjust exit timing

ngshijun committed:
Extract out individual method for different user inputs for the parse() method in Parser class

The method parse() in Parser class in too long and might cause confusion to others who will be reading it.

Extracting out individual method for different input command.

This will makes the code abiding to the Single Level of Abstraction Principle (SLAP), so there's only one level of abstraction per function.

ngshijun committed:
Merge pull request #2 from ngshijun/branch-A-CodeQuality

Extract out individual method for different user inputs for the parse…

ngshijun committed:
Merge branch 'master' of https://github.com/ngshijun/ip

* 'master' of https://github.com/ngshijun/ip:
  Extract out individual method for different user inputs for the parse() method in Parser class

ngshijun committed:
Fix coding standard error

ngshijun committed:
Add assert to document important assumptions that should hold

ngshijun committed:
Merge pull request #4 from ngshijun/branch-A-Assertions

Add assert to document important assumptions that should hold

ngshijun committed:
Add font to .jar file

ngshijun committed:
Add extension

ngshijun committed:
Add user guide

ngshijun committed:
Add Ui.png image

ngshijun committed:
Update readme and Ui.png

ngshijun committed:
fix font issue

ngshijun committed:
update readme

ngshijun committed:
fix error in coding standard

ngshijun committed:
Edit readme

ngshijun committed:
fix storage issue

ngshijun committed:
Add title for the application

ngshijun committed:
fix emoji not shown
